util/types/time.go

```diff
@@ -673,6 +673,14 @@ type Duration struct {
 	Fsp int
 }
 
+//Add adds d to d, returns a duration value.
+func (d Duration) Add(v *Duration) Duration {
```

Comment:
I reference the code  
https://github.com/pingcap/tidb/blob/master/util/types/time.go#L464